src/ucp/rndv/rndv.c

```diff
@@ -1413,13 +1436,15 @@ UCS_PROFILE_FUNC(ucs_status_t, ucp_rndv_progress_am_bcopy, (self),
                                        ucp_rndv_pack_data,
                                        ucp_rndv_pack_data, 1);
     }
-    if (status == UCS_OK) {
-        ucp_rndv_complete_send(sreq, UCS_OK);
-    } else if (status == UCP_STATUS_PENDING_SWITCH) {
-        status = UCS_OK;
+
+    ret_status = ucp_am_bcopy_handle_status_from_pending(self, !single, 0, 0,
```

Comment:
This is the only place, where `no_complete` is passed to `ucp_am_bcopy_handle_status_from_pending`.
I'd remove changes in `ucp_am_bcopy_handle_status_from_pending` and rewrite code here without it

Reply:
done